drivers/flash/soc_flash_nrf.h

```diff
@@ -37,7 +37,7 @@ struct flash_context {
 			  (FLASH_PAGE_MAX_CNT) / 1000 * 15 / 10)
 #else
 
-#define FLASH_TIMEOUT_MS ((FLASH_PAGE_ERASE_MAX_TIME_US) * \
+#define FLASH_TIMEOUT_MS (2 * (FLASH_PAGE_ERASE_MAX_TIME_US) * \
```

Comment:
Depending on the number of other tickers active (BLE states/roles), the flash operations would round-robin with the rest of the tickers.

Have a suitably named Kconfig option, would address the `multiplied by 1.5`  used for `CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE` as well that is just above.